rust/lancedb/src/database/listing/manifest.rs

```diff
@@ -0,0 +1,3991 @@
+        format!("{:08x}_{}", (hash & 0xFFFFFFFF) as u32, object_id)
+    }
+
+    fn build_object_uri(
+        base_uri: &str,
+        dir_name: &str,
+        query_string: Option<&str>,
+    ) -> Result<String> {
+        let mut uri = base_uri.to_string();
+        if !uri.ends_with('/') {
+            uri.push('/');
+        }
+
+        uri.push_str(dir_name);
+
+        if let Some(query) = query_string {
+            uri.push('?');
+            uri.push_str(query);
+        }
+
+        Ok(uri)
+    }
+
+    async fn validate_namespace_levels_exist(&self, namespace: &[String]) -> Result<()> {
+        if !namespace.is_empty() {
+            for i in 1..=namespace.len() {
+                let ns_id = namespace[..i].join(DELIMITER);
+                if !self.manifest_contains_object(&ns_id).await? {
+                    return Err(Error::NamespaceNotFound { name: ns_id });
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn get_string_column<'a>(batch: &'a RecordBatch, column_name: &str) -> Result<&'a StringArray> {
+        batch
+            .column_by_name(column_name)
+            .ok_or_else(|| Error::Runtime {
+                message: format!("Missing {} column", column_name),
+            })?
+            .as_any()
+            .downcast_ref::<StringArray>()
+            .ok_or_else(|| Error::Runtime {
+                message: format!("Invalid {} type", column_name),
+            })
+    }
+
+    #[allow(dead_code)]
+    fn get_list_column<'a>(batch: &'a RecordBatch, column_name: &str) -> Result<&'a ListArray> {
+        batch
+            .column_by_name(column_name)
+            .ok_or_else(|| Error::Runtime {
+                message: format!("Missing {} column", column_name),
+            })?
+            .as_any()
+            .downcast_ref::<ListArray>()
+            .ok_or_else(|| Error::Runtime {
+                message: format!("Invalid {} type", column_name),
+            })
+    }
+
+    /// Register a table in the manifest without creating the physical table
+    /// TODO: this could be in public database API
+    pub async fn register_table(
+        &self,
+        name: &str,
+        namespace: &[String],
+        location: String,
+    ) -> Result<()> {
+        self.ensure_manifest_table_exists().await?;
+
+        validate_table_name(name)?;
+        validate_namespace(namespace)?;
+        self.validate_namespace_levels_exist(namespace).await?;
+
+        let object_id = Self::build_object_id(namespace, name);
+        if self.manifest_contains_object(&object_id).await? {
+            return Err(Error::TableAlreadyExists {
+                name: name.to_string(),
+            });
+        }
+
+        self.insert_into_manifest(object_id, ObjectType::Table, Some(location), None, None)
+            .await
+    }
+
+    /// List all tables in a namespace with their metadata
+    /// TODO: this could be in public database API
```

Comment:
What does this todo comment mean?  Don't we already have a list_tables in the `Database` trait?

Reply:
this one contains table location information because the location is no longer hard coded from table name, the current one is just `table_names`.